### PR TITLE
Refactoring and changes to mega menu JavaScript

### PR DESCRIPTION
**Pull request by punalsc**
Hi @gtvj ,

Could you have a look at this PR, and let me know your thoughts.

- [x] Extend Grunt task so that it runs QUnit tests at different browser widths
- [x] Tests for necessary classes on test fixture (and confirm that test fixture matches output from mega-menu-feed-processor application)
- [x] Tests that mega menu JavaScript behaviour works at correct screen sizes
- [x] Implement 'plus' and 'minus' styling for menu items

Thanks.

### CODE REVIEW

**Comment by punalsc** [comment]
Hi @gtvj ,

I have finished point 2 that tests/checks the necessary classes and fixtures. Could you have a look this the PR, and let me know if i need to change anything.
Thanks

**Comment by punalsc** [comment]
Hi @gtvj ,
I have implemented the minus and plus on toggle with tests. Could you have a look a look through when you can.
Thanks

File: css/base-sass.css [hunk 1]
```diff
@@ -5080,7 +5080,7 @@ blockquote:before {
       display: block; } }
 
 .minus {
-  background: url(../img/minus.png) 5px 12px no-repeat !important;
```
Comment (gtvj) [inline css/base-sass.css]: @punalsc I think we should address this with specificity rather than an `!important` rule. I also wonder if `expanded` might be a more descriptive name than `minus`.

File: css/sass/_mega-menu.scss [hunk 2]
```diff
@@ -65,7 +65,7 @@
 }
 
 .minus {
-  background: url(../img/minus.png) 5px 12px no-repeat !important;
```
Comment (gtvj) [inline css/sass/_mega-menu.scss]: @punalsc I think we should address this with specificity rather than an `!important` rule. The [development guide](https://github.com/nationalarchives/development-guide-and-peer-reviews/blob/master/development-guide.md) says that we should avoid using `!important`

I also wonder if `expanded` might be a more descriptive name than `minus`.

File: js/compiled/tna-base.js [hunk 3]
```diff
@@ -85,6 +85,14 @@ $.fn.webtrends_click_handler = function () {
     })
 };
 
+//Toggles class of minus when clicked.
+$.fn.toggleMinusClass = function (){
```
Comment (gtvj) [inline js/compiled/tna-base.js]: @punalsc - this doesn't seem to have been implemented correctly. As a plugin it should act upon the jQuery collection to which it is applied. The way it has been implemented the function body makes no reference to the jQuery collection and it instead defines a selector within the function body.

File: js/compiled/tna-base.js [hunk 4]
```diff
@@ -294,6 +302,8 @@ $(".mega-menu > ul > li > a").mega_menu_enhancements();
 
 $('a', '.mega-menu').webtrends_click_handler();
 
+$('.mega-menu > ul > li > a').toggleMinusClass();
```
Comment (gtvj) [inline js/compiled/tna-base.js]: @punalsc - see comment above.

File: css/sass/_header.scss [hunk 5]
```diff
@@ -129,23 +129,20 @@ header {
...
+          margin-right: auto;
+          margin-left: auto;
+          max-width: 83%;
+      }
     }
   }
-  .search-expander {
-    height: 2.1em;
+  .search-expander, .search-close {
+    height: 2.7em;
```
Comment (gtvj) [inline css/sass/_header.scss]: Hi @punalsc - why the change to `.search-expander` height? Has something changed?
